.github/workflows/auto_publish_release.yml

```diff
@@ -0,0 +1,83 @@
+name: Release stage 2 - auto publish
+
+on:
+  pull_request:  # types AND paths
+    types: [closed]
+    paths: ['package.json']
+    # NOTE: While this is too generic, we use the `if` condition of the job to narrow it down
+    # NOTE: Don't use `branches` as we might create release on any branch
+
+env:
+  # Best not to include the GH token here, only do it for the steps that need it
+  MERGE_SHA: ${{ github.event.pull_request.merge_commit_sha }}
+  CHANGELOG_FILE: CHANGES.md
+
+jobs:
+  publish:
+    if: >-  # NOTE: Can't use top-level env here unfortunately
+      github.event.pull_request.merged == true &&
+      contains(github.event.pull_request.labels.*.name, 'release')
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ env.MERGE_SHA }}
+
+      - name: Install
+        run: yarn install
+
+      - name: Build
+        run: yarn build --optimize
+
+      - name: Get the version number
+        run: |
+          VERSION=$(node -p -e "require('./package.json').version")
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+
+      - name: Create ZIP archive
+        run: |
+          cd dist && zip -r ../cylc-ui-${{ env.VERSION }}-dist.zip *
+
+      - name: sha256sum
+        run: |
+          SHA256SUM=$(sha256sum cylc-ui-${{ env.VERSION }}-dist.zip)
+          echo "SHA256SUM=$SHA256SUM" >> $GITHUB_ENV
+
+      - name: Publish GitHub release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          commitish: ${{ env.MERGE_SHA }}
+          tag_name: ${{ env.VERSION }}
+          release_name: cylc-ui ${{ env.VERSION }}
+          prerelease: ${{ env.PRERELEASE }}
```

Comment:
I don't think this will work as `$PRERELEASE` is not set? In the Python project workflows, I've used the Packaging module to determine if the release is pre or not in the "Get version number" step. Might be possible using a node/npm tool. Or can just get rid of this argument/option.

Reply:
Ah, good catch. I had a look at the `npm version` command, and whether we could find out whether the selected version is a prerelease or not. But found nothing ready to use, the closest I found were regexes, but probably better to look into that later and just remove that option for now IMO.